src/main/java/de/maxhenkel/audioplayer/command/AudioPlayerCommands.java

```diff
@@ -399,15 +400,15 @@ private static LiteralArgumentBuilder<CommandSourceStack> bulkApplyCommand(Liter
                                     ItemStack itemInHand = player.getItemInHand(InteractionHand.MAIN_HAND);
                                     String customName = StringArgumentType.getString(context, "custom_name");
                                     if (validator.test(itemInHand)) {
-                                        processShulker(context, itemInHand, sound, customName);
+                                        processShulker(context, itemInHand, itemTypeName, sound, customName);
                                     } else {
                                         context.getSource().sendFailure(Component.literal("You don't have a %s in your main hand".formatted(itemTypeName)));
                                     }
                                     return 1;
                                 })));
     }
 
-    private static void processShulker(CommandContext<CommandSourceStack> context, ItemStack itemInHand, UUID soundID, @Nullable String name) {
+    private static void processShulker(CommandContext<CommandSourceStack> context, ItemStack itemInHand, String itemTypeName, UUID soundID, @Nullable String name) {
         ListTag shulkerContents = itemInHand.getTagElement("BlockEntityTag").getList("Items", 10);
```

Comment:
`itemInHand.getTagElement` might return `null` if it doesn't exist

Reply:
Yeah I wasn't sure the best way to handle that. We know that it won't be null because it has been verified there's a shulker box in the hand and it has to have the tag... but looks like suggested fix is wrapping the call in `Objects.requireNonNull()`.

Uploaded. Please guide if there's a better option

Reply:
Switched to `.getOrCreateTagElement` per other comments.